test/unit/cmap/connect.test.ts

```diff
@@ -185,9 +185,156 @@ describe('Connect Tests', function () {
+        });
+
+        it(`should not have 'name' property in client.env `, async () => {
+          const handshakeDocument = await prepareHandshakeDocument(authContext);
+          expect(handshakeDocument.client.env).to.not.have.property('name');
+        });
+
+        context('when 512 byte size limit is exceeded', async () => {
+          it(`should not 'env' property in client`, async () => {
+            let longAppName = '';
+            // make metadata = 507 bytes, so it takes up entire LimitedSizeDocument
+            for (let i = 0; i < 493; i++) {
+              longAppName += 's';
+            }
+            const longAuthContext = {
+              connection: {},
+              options: { ...CONNECT_DEFAULTS, metadata: { appName: longAppName } }
+            };
+            const handshakeDocument = await prepareHandshakeDocument(longAuthContext);
+            expect(handshakeDocument.client).to.not.have.property('env');
+          });
+        });
+      });
+
+      context('when kubernetes and FAAS are both present', () => {
+        const authContext = {
+          connection: {},
+          options: { ...CONNECT_DEFAULTS, metadata: { env: { name: 'aws.lambda' } } }
+        };
+
+        beforeEach(() => {
+          process.env.KUBERNETES_SERVICE_HOST = 'I exist';
+        });
+
+        afterEach(() => {
+          if (cachedEnv.KUBERNETES_SERVICE_HOST != null) {
+            process.env.KUBERNETES_SERVICE_HOST = cachedEnv.KUBERNETES_SERVICE_HOST;
+          } else {
+            delete process.env.KUBERNETES_SERVICE_HOST;
+          }
+        });
+
+        it(`should include { orchestrator: 'kubernetes'} in client.env.container`, async () => {
+          const handshakeDocument = await prepareHandshakeDocument(authContext);
+          expect(handshakeDocument.client.env.container.orchestrator).to.equal('kubernetes');
+        });
+
+        it(`should still have properly set 'name' property in client.env `, async () => {
+          const handshakeDocument = await prepareHandshakeDocument(authContext);
+          expect(handshakeDocument.client.env.name).to.equal('aws.lambda');
+        });
+
+        context('when 512 byte size limit is exceeded', async () => {
+          it(`should not have 'container' property in client.env`, async () => {
+            let longAppName = '';
+            // make metadata = 507 bytes, so it takes up entire LimitedSizeDocument
+            for (let i = 0; i < 485; i++) {
+              longAppName += 's';
+            }
+            const longAuthContext = {
+              connection: {},
+              options: {
+                ...CONNECT_DEFAULTS,
+                metadata: {
+                  appName: longAppName,
+                  env: { name: 'aws.lambda' }
+                }
+              }
+            };
+            const handshakeDocument = await prepareHandshakeDocument(longAuthContext);
+            expect(handshakeDocument.client).to.not.have.property('env');
```

Comment:
The test name is "should not have 'container' property in client.env" but we're asserting there is no "env" at all. I don't think that is what we would expect here, I think the input metadata you've got here is already above 512 which shouldn't be possible real-world since we build the metadata with a LimitedSizeDocument of 512 in parseOptions.

Reply:
My mistake, I need to double check my copy pasting.